src/Adapter/Product/SpecificPrice/CommandHandler/AddProductSpecificPriceHandler.php

```diff
@@ -0,0 +1,108 @@
+use PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\Command\AddProductSpecificPriceCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\CommandHandler\AddProductSpecificPriceHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\SpecificPrice\Exception\SpecificPriceConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\SpecificPrice\ValueObject\SpecificPriceId;
+use PrestaShop\PrestaShop\Core\Util\DateTime\DateTime;
+use PrestaShopException;
+use SpecificPrice;
+
+/**
+ * Handles AddProductSpecificPriceCommand using legacy object model
+ */
+final class AddProductSpecificPriceHandler implements AddProductSpecificPriceHandlerInterface
+{
+    /**
+     * @var SpecificPriceRepository
+     */
+    private $specificPriceRepository;
+
+    /**
+     * @param SpecificPriceRepository $specificPriceRepository
+     */
+    public function __construct(SpecificPriceRepository $specificPriceRepository)
+    {
+        $this->specificPriceRepository = $specificPriceRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(AddProductSpecificPriceCommand $command): SpecificPriceId
+    {
+        $specificPrice = $this->createSpecificPriceFromCommand($command);
+
+        return $this->specificPriceRepository->add($specificPrice);
+    }
+
+    /**
+     * Creates legacy SpecificPrice object from command
+     *
+     * @param AddProductSpecificPriceCommand $command
+     *
+     * @return SpecificPrice
+     *
+     * @throws PrestaShopException
+     * @throws SpecificPriceConstraintException
+     */
+    private function createSpecificPriceFromCommand(AddProductSpecificPriceCommand $command): SpecificPrice
+    {
+        $specificPrice = new SpecificPrice();
+
+        $specificPrice->id_product = $command->getProductId()->getValue();
+        $specificPrice->reduction_type = $command->getReduction()->getType();
+        $specificPrice->reduction = $command->getReduction()->getValue();
+        $specificPrice->reduction_tax = $command->includesTax();
+        $specificPrice->price = $command->getPrice();
+        $specificPrice->from_quantity = $command->getFromQuantity();
+        $specificPrice->id_shop_group = $command->getShopGroupId() ?? 0;
+        $specificPrice->id_shop = $command->getShopId() ?? 0;
+        $specificPrice->id_product_attribute = null !== $command->getCombinationId() ? $command->getCombinationId()->getValue() : 0;
+        $specificPrice->id_currency = $command->getCurrencyId() ?? 0;
+        $specificPrice->id_country = $command->getCountryId() ?? 0;
+        $specificPrice->id_group = $command->getGroupId() ?? 0;
+        $specificPrice->id_customer = $command->getCustomerId() ?? 0;
+        $specificPrice->from = DateTime::NULL_VALUE;
+        $specificPrice->to = DateTime::NULL_VALUE;
+
+        $from = $command->getDateTimeFrom();
+        if ($from) {
+            $specificPrice->from = $from->format('Y-m-d H:i:s');
```

Comment:
```suggestion
            $specificPrice->from = $from->format(DateTime::DEFAULT_DATETIME_FORMAT);
```

from `PrestaShop\PrestaShop\Core\Util\DateTime\DateTime`